File: socialgo/src/main/java/com/fungo/socialgo/listener/OnLoginListener.java
package com.fungo.socialgo.listener;

import com.fungo.socialgo.exception.SocialError;
import com.fungo.socialgo.model.LoginResult;

/**
 * CreateAt : 2016/12/25
 * Describe : 登陆监听
 *
 * @author chendong
 */
public interface OnLoginListener {

    void onStart();

    void onSuccess(LoginResult loginResult);

    void onCancel();

    void onFailure(SocialError e);
}

File: socialgo/src/main/java/com/fungo/socialgo/listener/OnLoginListener.kt
package com.fungo.socialgo.listener

import com.fungo.socialgo.exception.SocialError
import com.fungo.socialgo.model.LoginResult

/**
 * 登录回调
 */
interface OnLoginListener {

    fun onStart()

    fun onSuccess(loginResult: LoginResult)

    fun onCancel()

    fun onFailure(e: SocialError)
}
